assets/chat.js

```diff
@@ -1,30 +1,18 @@
 import Chat from "./chat/js/chat";
 import emotes from "./emotes.json";
 
-$.when(
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/me`)
-            .done(res)
-            .fail(() => res(null))
-    ),
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/history`)
-            .done(res)
-            .fail(() => res(null))
-    ),
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/viewer-states`)
-            .done(res)
-            .fail(() => res([]))
-    )
-).then(
-    (userAndSettings, history, viewerStates) =>
-        (window.__chat__ = new Chat()
-            .withUserAndSettings(userAndSettings)
-            .withEmotes(emotes)
-            .withGui()
-            .withViewerStates(viewerStates)
-            .withHistory(history)
-            .withWhispers()
-            .connect(WEBSOCKET_URI))
-);
+const init = async v => {
+    const userInfo = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/me`).done(res).fail(() => res(null)));
+    const history = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/history`).done(res).fail(() => res(null)));
+    const viewerStates = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/viewer-states`).done(res).fail(() => res(null)));
+
+    const promiseSync = await Promise.all([userInfo(), history(), viewerStates()]);
+    
+    const chat = new Chat();
+    window.__chat__ = chat; // I do not know why this is needed but included it if an app depends on it
+    await chat.withGui()
+    .then(async() => await Promise.all([chat.withUserAndSettings(promiseSync[0]), chat.withEmotes(emotes), chat.withViewerStates(promiseSync[2]), chat.withHistory(promiseSync[1]), chat.withWhispers()]))
```

Comment:
adding async to functions that do not perform async operations does not make them run in parallel.

Reply:
Pardon, you are correct and I knew about that but only an updated version of whispers would benefit and the rest would remain unaffected. I did not know about when doing the same.